### PR TITLE
add design document to refactor port config logic

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**

### CODE REVIEW

File: doc/port-config-refactor/port-config-refactor-design.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# SONiC Port Configuration Refactor Design #
```
Comment (keboliu) [inline doc/port-config-refactor/port-config-refactor-design.md:1]: I think the changing design part is OK, suggest adding a test plan section

Comment (Junchao-Mellanox) [reply doc/port-config-refactor/port-config-refactor-design.md:1]: Added test plan part, please review

File: doc/port-config-refactor/port-config-refactor-design.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (keboliu) [inline doc/port-config-refactor/port-config-refactor-design.md:2]: https://github.com/Azure/sonic-platform-common/pull/72   check this PR, not quite sure whether it's already covered some change?

Comment (Junchao-Mellanox) [reply doc/port-config-refactor/port-config-refactor-design.md:2]: I don't understand that PR, it said it support platform.json, but I don't see any code parse platform.json

Comment (keboliu) [reply doc/port-config-refactor/port-config-refactor-design.md:2]: then we ignor, I just thought maybe the also made some change.

**Review by keboliu — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
Community PR: https://github.com/Azure/SONiC/pull/613

**Pull request closed without merging**